### PR TITLE
[WIP] Add Anaconda

### PR DESCRIPTION
**Pull request by adidalal**
Inspired by #13359, I decided to try to add [Anaconda](https://store.continuum.io/cshop/anaconda/) to `brew cask`.

The first issue is that it's actually packaged as a shell script. (ie. you run `bash Anaconda3-2.3.0-MacOSX-x86_64.sh -b` to install via command-line silently, and I'm not sure what stanza would do that)

It installs to `~/anaconda3` by default, and I'm not sure that is expected behavior for cask. (Note that we can change the default location by passing `-p {path) to it', but with the upcoming change, should we restrict things to only the Caskroom directory, anyway?

It adds itself to the path (ie. `export PATH="/Users/aditya/anaconda3/bin:$PATH"`) - would this be a caveat?

There is a GUI interface at `~/anaconda3/Launcher.app` as well that should be linked.

Help appreciated, thanks!

### CODE REVIEW

**Comment by vitorgalvao** [comment]
> but with the upcoming change

That change will mimic usual install behaviour, so we’ll do whatever default installation this does.

> It adds itself to the path

What does it do that, exactly, does it append a line to shell startup files? If yes, which ones?

> There is a GUI interface at `~/anaconda3/Launcher.app` as well that should be linked.

Try `app`, as usual.

To install this, take a look at [postflight](https://github.com/caskroom/homebrew-cask/blob/master/doc/CASK_LANGUAGE_REFERENCE.md#flight-mini-dsl).

What kind of an idea is it to package your app as a 257MB shell script, anyway?

**Comment by adityadalal924** [comment]
The path is added into `.bashrc`, so maybe a caveat like:

```
caveats do
    <<-EOS.undent
    To use Anaconda, you must add the $ANACONDA_ROOT/bin directory
    to your PATH environment variable.
    For bash shell, add these lines to ~/.bash_profile:

      export ANACONDA_ROOT=~/anaconda3
      export PATH=$ANACONDA_ROOT/bin:"$PATH"
    
EOS
end
```

Linking the .app is fine (and makes sense), but how do you run `bash Anaconda3-2.3.0-MacOSX-x86_64.sh -b` from within brew (is there an exec command).

I came up with

```
  installer :script => 'Anaconda3-#{version}-MacOSX-x86_64.sh',
            :args => %w{-b},
            :sudo => false
```
 does that seem okay?



The install script is [here](https://gist.github.com/adityadalal924/81a550ade6d2375a56dc) FYI - everything is packaged as a binary blob after the script.

**Comment by vitorgalvao** [comment]
Oh, right, I forgot about `installer :script`. Yes, that looks OK.

**Comment by adityadalal924** [comment]
Still no luck, would you mind taking a look? Thanks

File: Casks/anaconda.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  sha256 'c4bb59a57bf44dde80612041bbbcfd2e5cab8534842209ef456da7a46f919c33'
+
+  url "https://repo.continuum.io/archive/Anaconda-#{version}-MacOSX-x86_64.sh"
+  name 'Anaconda'
+  homepage 'https://store.continuum.io/cshop/anaconda/'
+  license :gratis
+  tags :vendor => 'Continuum Analytics'
+
+  installer :script => 'Anaconda-#{version}-MacOSX-x86_64.sh',
+            :args => [ '-b', '-p', '#{staged_path}' ]
```
Comment (vitorgalvao) [inline Casks/anaconda.rb]: Remove `, '-p', '#{staged_path}'`. Don’t give it an install path, let it use the default.

Similarly, an `uninstall` must be added, removing said path.

File: Casks/anaconda.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  app 'Launcher.app'
+
+  depends_on :macos => '>= :lion'
+
+  caveats do
```
Comment (vitorgalvao) [inline Casks/anaconda.rb]: No need for `do…end`. Make it instead:

```ruby
  caveats <<-EOS.undent
    To use Anaconda from the command line, you must add the   
    $ANACONDA_ROOT/bin directory to your PATH environment variable.
    For bash shell, add these lines to ~/.bash_profile:

      export ANACONDA_ROOT=#{staged_path}
      export PATH=$ANACONDA_ROOT/bin:"$PATH"
  EOS
```

**Comment by vitorgalvao** [comment]
What’s the difficulty you’re having?

**Comment by adityadalal924** [comment]
I run into this error:

```
Error: Uh oh, could not identify primary container for '/Library/Caches/Homebrew/anaconda-2.3.0.sh'
```

It needs to be executed as `bash Anaconda-2.3.0-MacOSX-x86_64.sh -b`, any thoughts on getting that working?

The `app: 'Launcher.app'` stanza probably won't work, because the full path is `~/anaconda/Launcher.app` (not related to any caskroom directories - that's why I tried using staged_path)

Updated caveats & added uninstall.

File: Casks/anaconda.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  version '2.3.0'
+  sha256 'c4bb59a57bf44dde80612041bbbcfd2e5cab8534842209ef456da7a46f919c33'
+
+  url "https://repo.continuum.io/archive/Anaconda-#{version}-MacOSX-x86_64.sh"
+  name 'Anaconda'
+  homepage 'https://store.continuum.io/cshop/anaconda/'
+  license :gratis
+  tags :vendor => 'Continuum Analytics'
+
+  installer :script => 'Anaconda-#{version}-MacOSX-x86_64.sh',
```
Comment (vitorgalvao) [inline Casks/anaconda.rb]: This needs to use double quotes. `#{version} isn’t being expanded. Needs to be `installer :script => "Anaconda-#{version}-MacOSX-x86_64.sh",`

File: Casks/anaconda.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            :args => [ '-b' ]
```
Comment (vitorgalvao) [inline Casks/anaconda.rb]: This does not need to be run as root (and should’t), so we need to add `:sudo => false`.

File: Casks/anaconda.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+  app 'Launcher.app'
```
Comment (vitorgalvao) [inline Casks/anaconda.rb]: It seems like a link to it is set on the Desktop as part of the installation, so lets leave this out.

**Comment by vitorgalvao** [comment]
> Error: Uh oh, could not identify primary container for '/Library/Caches/Homebrew/anaconda-2.3.0.sh'

[This line](https://github.com/caskroom/homebrew-cask/blob/99bf6412058bdcc17e23171fc283b3fce7b4c499/Casks/dogestry.rb#L11) fixes it.

**Comment by vitorgalvao** [comment]
Thank you for the contribution. It needed some fixes, so they were made in a5e7b4345a2c947a198db21448c4e600a7c0a567. Your contribution is still included (and still credited to you), with the appropriate modifications. Please feel free to ask about any of the changes.

**Comment by adityadalal924** [comment]
Excellent, thanks!

**Pull request closed without merging**